sympy/core/relational.py

```diff
@@ -345,7 +345,7 @@ def __new__(cls, lhs, rhs=0, **options):
                                 rv = fuzzy_bool(Eq(*args))
                                 if rv is True:
                                     rv = None
-                elif any(a.is_infinite for a in Add.make_args(n)):  # (inf or nan)/x != 0
```

Comment:
I think this is not the right place to catch your condition. The intent here is to return False for `Eq(foo-oo,0)` since there is no `foo` that can make the lhs 0. It's better to do your test with the arguments, perhaps up where `fin` is defined: 

```
if None not in fin: 
    if L != R: 
        return S.false 
    if L is False: 
        return S.true 
elif False in fin and None in fin:  # something like this...
    return None
```

Or maybe that test needs to happen if there is no denominator...Not sure and don't have time to think this out right now.